src/InvalidArgument.php

```diff
@@ -0,0 +1,8 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT;
+
+final class InvalidArgument extends Exception
```

Comment:
To reduce the BC surface, this should extend `InvalidArgumentException`

Reply:
The main goal here is to not depend on SPL exception class. I'm extending this class from the Exception class of this library.

We can still do BC break since we are only in alpha stage.

Reply:
I'm sorry that my poor explanation led to this small confusion. The scope of the problem isn't the dependence on SPL but rather the fact that we directly expose it.

We're surely handling a new major release but that doesn't mean we **need** to break the BC everywhere. This kind of change is definitely one that can be compatible. and should IMHO.